XYZDrivers/src/main/java/com/xyzdrivers/repositories/ClaimsRepo.java

```diff
@@ -58,34 +67,64 @@ public void delete(Claim model) throws RepositoryException {
     }
 
     @Override
-    public void insert(Claim model) throws RepositoryException {
-        throw new UnsupportedOperationException();
-    }
+    public void insert(Claim c) throws RepositoryException {
+        Object[] splitClaim = new Object[5];
+
+        try {
+            if (c == null) {
+                throw new IllegalArgumentException("The object c is null.");
```

Comment:
Don't catch this. RepositoryException should be used as a catch-all for exceptions in repositories that weren't the caller's fault, in this case the caller provided a null argument so it's entirely their fault and their responsibility to deal with the consequences

Reply:
IllegalArgumentException isn't caught here, it's left to bubble up to the caller.

Reply:
It's caught on line 122